### ISSUE TITLE
Support persisted version in original pom

### ISSUE THREAD

**Issue by reitzmichnicht** [qoomon/maven-git-versioning-extension#33]
The plugin is really working great for feature branches, but there is one feature missing that would be useful for release branches.

Having a release branch the plugin could automatically determine the version from the branch name, but we also want to commit the sources with this version.
So would be great to have a optional setting in maven-git-versioning-extension.xml that the versioning is not done in a temporary file like for feature branches but in the original pom.

Kind regards,
Michael

**Comment by qoomon** [comment]
hi, do you think of an `mvn` parameter like `-DversioningPersist=true`?
Or do you want persist the version only on specific branches?

Another way would be the following code lines
```shell
# get project version
project_version=$(mvn --non-recursive exec:exec -Dexec.executable='echo' -Dexec.args='${project.version}' -q)
# set pom version
mvn versions:set -DnewVersion=$project_version
```

**Comment by reitzmichnicht** [comment]
I would love a configuration by branch. I know i could do it with several maven calls like you pointed out, but having this in the branch configuration would be awesome.

**Comment by qoomon** [comment]
Okay I think it's possible, but I need some time to figuring out how `mvn versions:set` does its job or if it's possible to start `mvn versions:set` from this extension

**Comment by qoomon** [comment]
I will have a close look at this [post](x https://massol.myxwiki.org/xwiki/bin/view/Blog/Calling%20a%20Maven%20plugin%20from%20another%20plugin) maybe I could call the `versions:set` plugin like this

**Comment by qoomon** [comment]
Do anybody know how to update a XML file in Java without loosing formatting of the original one?

**Comment by reitzmichnicht** [comment]
Reusing the versions plugin is a reasonable idea, but to my understanding most of the work for setting versions should be already part of your plugin.

Jgitflow plugin also allows to replace versions in pom.xml, maybe you find this useful:
https://bitbucket.org/atlassian/jgit-flow/src/831bc353594a3633ca5c7144858a25b03fa26ba2/jgitflow-maven-plugin/src/main/java/com/atlassian/maven/plugins/jgitflow/rewrite/MavenProjectRewriter.java?at=develop&fileviewer=file-view-default

**Comment by qoomon** [comment]
Thanks for this hint, however it is quite quirky solution :-D

**Comment by qoomon** [comment]
first step is done see f641ee3c54cf491dd84f9e21de70e1246058c063

**Comment by qoomon** [comment]
any suggestions for the option name?

**Comment by reitzmichnicht** [comment]
updatePom sounds good :+1:

**Comment by qoomon** [comment]
just released v4.1.0
